src/seedu/addressbook/data/person/UniquePersonList.java

```diff
@@ -33,46 +31,27 @@ protected DuplicatePersonException() {
     private final List<Person> internalList = new ArrayList<>();
 
     /**
-     * Constructs empty PersonList.
+     * Constructs empty person list.
      */
     public UniquePersonList() {}
 
     /**
-     * Varargs/array constructor, enforces no nulls or duplicates.
-     */
-    public UniquePersonList(Person... persons) throws DuplicatePersonException {
-        Utils.assertNotNull(persons);
-        final List<Person> initialPersons = Arrays.asList(persons);
-        if (!Utils.elementsAreUnique(initialPersons)) {
-            throw new DuplicatePersonException();
-        }
-        internalList.addAll(initialPersons);
-    }
-
-    /**
-     * java collections constructor, enforces no null or duplicate elements.
+     * Constructs a list from the items in the given collection.
+     * @param persons a collection of persons
+     * @throws DuplicatePersonException if the {@code persons} contains duplicate persons
      */
     public UniquePersonList(Collection<Person> persons) throws DuplicatePersonException {
-        Utils.assertNoNullElements(persons);
         if (!Utils.elementsAreUnique(persons)) {
             throw new DuplicatePersonException();
         }
         internalList.addAll(persons);
     }
 
     /**
-     * java set constructor, enforces no nulls.
-     */
-    public UniquePersonList(Set<Person> persons) {
-        Utils.assertNoNullElements(persons);
-        internalList.addAll(persons);
-    }
-
-    /**
-     * Copy constructor, insulates from changes in source.
+     * Constructs a shallow copy of the list.
      */
     public UniquePersonList(UniquePersonList source) {
-        internalList.addAll(source.internalList); // insulate internal list from changes in argument
+        internalList.addAll(source.internalList);
```

Comment:
So the constructor can access private variables of another object of the same class? Seems odd :-p TIL